### PR TITLE
Check catalog source status

### LINKED ISSUE #448: Check CatalogSource status in OperandRegistry controller to make sure the registry phase
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by DanielXLee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #448

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/controller/operandregistry/operandregistry_controller.go [hunk 1]
```diff
@@ -51,6 +54,11 @@ func Add(mgr manager.Manager) error {
 
 // newReconciler returns a new reconcile.Reconciler
 func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	// Add extenal api scheme
```
Comment (horis233) [inline pkg/controller/operandregistry/operandregistry_controller.go:57]: @DanielXLee Do we need to add the external api scheme here?
I think https://github.com/IBM/operand-deployment-lifecycle-manager/blob/master/pkg/apis/addtoscheme_operator_v1alpha1.go#L28 is enough to use OLM api.

File: pkg/controller/operandregistry/operandregistry_controller.go [hunk 2]
```diff
@@ -121,28 +129,8 @@ func (r *ReconcileOperandRegistry) Reconcile(request reconcile.Request) (reconci
...
-		instance.UpdateRegistryPhase(operatorv1alpha1.RegistryInit)
-	} else {
-		instance.UpdateRegistryPhase(operatorv1alpha1.RegistryRunning)
-	}
-	if err := r.client.Status().Update(context.TODO(), instance); err != nil {
-		return reconcile.Result{}, err
-	}
-
 	// Set Finalizer for the OperandRegistry
-	added := instance.EnsureFinalizer()
```
Comment (horis233) [inline pkg/controller/operandregistry/operandregistry_controller.go:144]: I wrote the condition in this way just want to make the code more readable.
I am ok with the change.
Could we enhance the comment to make it more clear?
For example
```
// // Set Finalizer for the OperandRegistry. If the OperandRegistry finalizer is added to the finalizer list, EnsureFinalizer() will return true. If the OperandRegistry finalizer already exists, EnsureFinalizer() will return false.
```

Comment (DanielXLee) [reply pkg/controller/operandregistry/operandregistry_controller.go:144]: Updated

File: pkg/controller/operandregistry/operandregistry_controller.go (hunk 1 above)
Comment (DanielXLee) [reply pkg/controller/operandregistry/operandregistry_controller.go]: you are right, I'll remove it

File: pkg/apis/operator/v1alpha1/operandregistry_types.go [hunk 3]
```diff
@@ -240,8 +246,25 @@ func (r *OperandRegistry) GetAllReconcileRequest() []reconcile.Request {
...
 
-func init() {
-	SchemeBuilder.Register(&OperandRegistry{}, &OperandRegistryList{})
+// SetReadyCondition creates a Condition to claim Ready
+func (r *OperandRegistry) SetReadyCondition(name string, rt ResourceType, cs corev1.ConditionStatus) {
+	c := newCondition(ConditionReady, cs, string(rt)+" is ready", string(rt)+" "+name+" is ready")
+	r.setCondition(*c)
+}
+
+// SetNoneCondition creates a Condition to claim Ready
```
Comment (chenzhiwei) [inline pkg/apis/operator/v1alpha1/operandregistry_types.go:255]: line 249 and 255 are same, is this the expected?

Comment (DanielXLee) [reply pkg/apis/operator/v1alpha1/operandregistry_types.go:255]: These are 2 different condition setup method, seems the comment need update

**Pull request merged**